service/server.py

```diff
@@ -55,26 +56,49 @@ def healthcheck():
 
 
 def get_property_address(postcode):
-    client = Elasticsearch([ELASTIC_SEARCH_ENDPOINT])
-    search = Search(
-        using=client, index='landregistry', doc_type='property_by_postcode_2')
+    search = create_search('property_by_postcode_2')
     query = search.filter('term', postcode=postcode)
     return query.execute().hits
 
 
+def get_properties_for_address(address):
+    search = create_search('property_by_address')
+    address_parts = address.split()
+    # In the future we might start weighting some words higher than others
+    # eg "Street" be low, if it is a structured address the house number should be high etc
+    word_queries = [~Q('match', address_string=address_part) for address_part in address_parts]
+    bool_address_query = Q('bool', should=word_queries)
+    query = search.query(bool_address_query)
+    return query.execute().hits
```

Comment:
according to the elasticsearch_dsl documentation, it needs to be done on the search object -> http://elasticsearch-dsl.readthedocs.org/en/latest/search_dsl.html